### PR TITLE
feat: updating sso provider end-point to be public

### PR DESCRIPTION
**Pull request by akoserwal**
http://localhost:8000/api/kafkas_mgmt/v1/sso_providers
end-point is public now. Bearer token is not required.

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

**Comment by akoserwal** [comment]
@wtrocki @Rajagopalan-Ranganathan @carlesarnal @MikeEdgar @ricardozanini

File: internal/kafka/internal/api/public/api/openapi.yaml [hunk 1]
```diff
@@ -1304,8 +1302,6 @@ components:
         created_at: 2021-04-07T16:24:01+05:30
     SsoProviderExample:
       value:
```
Comment (wtrocki) [inline internal/kafka/internal/api/public/api/openapi.yaml:1304]: This is not needed
```suggestion

```

File: internal/kafka/internal/api/public/api/openapi.yaml [hunk 2]
```diff
@@ -1304,8 +1302,6 @@ components:
...
-        kind: SsoProvider
-        href: /api/kafkas_mgmt/v1/sso_providers
         base_url: https://identity.api.redhat.com
         jwks: https://identity.api.openshift.com/auth/realms/rhoas/protocol/openid-connect/certs
         token_url: https://identity.api.openshift.com/auth/realms/rhoas/protocol/openid-connect/token
```
Comment (wtrocki) [inline internal/kafka/internal/api/public/api/openapi.yaml:1307]: WDYT about this proposal: https://github.com/redhat-developer/app-services-cli/pull/1565#discussion_r872842004

File: internal/kafka/internal/api/public/api/openapi.yaml (hunk 1 above)
Comment (akoserwal) [reply internal/kafka/internal/api/public/api/openapi.yaml:1304]: spec verification gives error. It is needed

**Review by carlesarnal — APPROVED** [review]
Love it, this is going to simplify our code in SR.

**Review by wtrocki — APPROVED** [review]

**Review by wtrocki — APPROVED** [review]

**Review by machi1990 — CHANGES_REQUESTED** [review]
Looks good to me from usage perspective.

Maybe we can update the integration test to test for this as well?
Removing https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f52a4fb36052b141d9bf27e8062780f07d293df6/internal/kafka/test/integration/serviceaccounts_test.go#L285-L286 as well as https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f52a4fb36052b141d9bf27e8062780f07d293df6/internal/kafka/test/integration/serviceaccounts_test.go#L303-L306 and passing an authenticated context might be enough. 
 
Wdyt @akoserwal ?

**Comment by akoserwal** [comment]
> Looks good to me from usage perspective.
> 
> Maybe we can update the integration test to test for this as well? Removing
> 
> https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f52a4fb36052b141d9bf27e8062780f07d293df6/internal/kafka/test/integration/serviceaccounts_test.go#L285-L286
> 
> as well as
> https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f52a4fb36052b141d9bf27e8062780f07d293df6/internal/kafka/test/integration/serviceaccounts_test.go#L303-L306
> 
> and passing an authenticated context might be enough.
> Wdyt @akoserwal ?

updated

**Review by machi1990 — APPROVED** [review]

**Review by MikeEdgar — APPROVED** [review]
lgtm

**Review by wtrocki — APPROVED** [review]

**Review by wtrocki — APPROVED** [review]

**Comment by akoserwal** [comment]
> > > Looks good to me from usage perspective.
> > > Maybe we can update the integration test to test for this as well? Removing
> > > https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f52a4fb36052b141d9bf27e8062780f07d293df6/internal/kafka/test/integration/serviceaccounts_test.go#L285-L286
> > > 
> > > as well as
> > > https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f52a4fb36052b141d9bf27e8062780f07d293df6/internal/kafka/test/integration/serviceaccounts_test.go#L303-L306
> > > 
> > > and passing an authenticated context might be enough.
> > > Wdyt @akoserwal ?
> > 
> > 
> > updated
> 
> Thanks @akoserwal it looks good to me.
> 
> Can you also add a verification of the `.Name` just like we are doing for the existing fields. The change can go in both
> 
> https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f52a4fb36052b141d9bf27e8062780f07d293df6/internal/kafka/test/integration/serviceaccounts_test.go#L309-L310
> 
> and
> https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f52a4fb36052b141d9bf27e8062780f07d293df6/internal/kafka/test/integration/serviceaccounts_test.go#L290-L291
> 
> Sorry, I missed mentioning it from the earlier round of review

updated

**Comment by wtrocki** [comment]
@machi1990 🙏 

image

**Review by machi1990 — APPROVED** [review]

**Comment by machi1990** [comment]
> @machi1990 pray
> 
> image

Sorry, I am only getting to this now. Merged already. 

/me love the meme game :-)

**Pull request merged**